core/src/autogluon/core/models/ensemble/bagged_ensemble_model.py

```diff
@@ -164,64 +162,37 @@ def _fit(self,
-            fold_start_n_repeat = fold_start + cur_repeat_count * k_fold
-            fold_end_n_repeat = min(fold_start_n_repeat + k_fold, fold_end)
-
-            for i in range(fold_start_n_repeat, fold_end_n_repeat):  # For each fold
-                fold_num_in_repeat = i - (j * k_fold)  # The fold in the current repeat set (first fold in set = 0)
-
-                fold_ctx = dict(
-                    model_name_suffix=f'S{j + 1}F{fold_num_in_repeat + 1}',  # S5F3 = 3rd fold of the 5th repeat set
-                    fold=kfolds[i],
-                    is_last_fold=i != (fold_end - 1),
-                    folds_to_fit=folds_to_fit,
-                    folds_finished=i - fold_start,
-                    folds_left=fold_end - i,
-                )
-
-                fold_fitting_strategy.schedule_fold_model_fit(model_base, fold_ctx, kwargs)
-            if (fold_end_n_repeat != fold_end) or (k_fold == k_fold_end):
-                self._k_per_n_repeat.append(k_fold)
-        fold_fitting_strategy.after_all_folds_scheduled()
-        self.models += models
-
-        self._bagged_mode = True
-
-        if self._oof_pred_proba is None:
-            self._oof_pred_proba = oof_pred_proba
-            self._oof_pred_model_repeats = oof_pred_model_repeats
+            self._fit_single(X=X, y=y, model_base=model_base, use_child_oof=use_child_oof, **kwargs)
+            return self
         else:
-            self._oof_pred_proba += oof_pred_proba
-            self._oof_pred_model_repeats += oof_pred_model_repeats
-
-        self._n_repeats = n_repeats
-        if k_fold == k_fold_end:
-            self._k = None
-            self._k_fold_end = 0
-            self._n_repeats_finished = self._n_repeats
-        else:
-            self._k = k_fold
-            self._k_fold_end = k_fold_end
-            self._n_repeats_finished = self._n_repeats - 1
+            refit_folds = self.params.get('refit_folds', False)
+            if refit_folds:
+                save_bag_folds = False
+                if kwargs.get('time_limit', None) is not None:
+                    fold_start = n_repeat_start * k_fold + k_fold_start
+                    fold_end = (n_repeats - 1) * k_fold + k_fold_end
+                    folds_to_fit = fold_end - fold_start
+                    # Reserve time for final refit model
+                    kwargs['time_limit'] = kwargs['time_limit'] * folds_to_fit / (folds_to_fit + 1.2)
+            self._fit_folds(X=X, y=y, model_base=model_base, k_fold=k_fold, k_fold_start=k_fold_start, k_fold_end=k_fold_end,
+                            n_repeats=n_repeats, n_repeat_start=n_repeat_start, save_folds=save_bag_folds, **kwargs)
+            # FIXME: Don't save folds except for refit
+            # FIXME: Cleanup self
+            # FIXME: Don't add `_FULL` to name
+            if refit_folds:
+                refit_template = self.convert_to_refit_full_template()
+                refit_template.params['use_child_oof'] = False
```

Comment:
nit: should this be in `convert_to_refit_full_template()` I assume this is what we always do on refit

Reply:
not necessarily because usually we don't directly hardcode the parent's OOF preds to the refit child, it is totally possible for the child to get its own OOF via LOO / efficient OOB. With that said, in future we may want to simplify and just always hardcode parent OOF to refit child OOF, as the nuance isn't necessarily all that helpful in practice.